### PR TITLE
vulkan: 1.2.189.1 -> 1.2.198.0

### LINKED ISSUE #147667: [vulkan] Out-of-date: 1.2.189.1 -> 1.2.200.
###### Checklist
- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: All packages in https://github.com/NixOS/nixpkgs/pull/141607.
current version: 1.2.189.1
desired version: 1.2.200

###### Notify maintainers
maintainers: @Ralith 

###### Note for maintainers
Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by Ralith**
Upstream has established rigorous tagging practices, allowing us to simplify things considerably.

###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/147667 and cleans up substantial duplication thanks to upstream tagging.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/vulkan-validation-layers/default.nix [hunk 1]
```diff
@@ -54,19 +22,22 @@ let
...
-  version = "1.2.189.1";
+  version = "1.2.198.0";
 
   # If we were to use "dev" here instead of headers, the setupHook would be
   # placed in that output instead of "out".
   outputs = ["out" "headers"];
   outputInclude = "headers";
 
-  src = (assert version == vulkan-headers.version;
+  src = (assert version == vulkan-headers.version
```
Comment (jansol) [inline pkgs/development/tools/vulkan-validation-layers/default.nix:32]: Could be nicer with something like
```nix
assert (lib.all (pkg: pkg.version == version) [vulkan-headers glslang spirv-tools spirv-headers])
```

**Review by jansol — COMMENTED** [review]
Glad to finally have some sense in this mess. Can we get this to work with nix-update now? Not sure how that works with packages that need to be upgraded in lockstep, might require a metapackage with an update script that updates all the individual packages?

Diff looks largely good to me but I did not try actually building.

File: pkgs/development/tools/vulkan-validation-layers/default.nix (hunk 1 above)
Comment (Ralith) [reply pkgs/development/tools/vulkan-validation-layers/default.nix]: Good idea; applied.

**Review by TilCreator — APPROVED** [review]
This change looks good too me, everything is much cleaner now. I also tested the new Glslang / Vulkan version with #152474 and it compiles and works just as expected.

**Pull request merged**